src/main/java/seedu/contax/logic/parser/FindAppointmentCommandParser.java

```diff
@@ -0,0 +1,49 @@
+package seedu.contax.logic.parser;
+
+import static seedu.contax.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_SEARCH_TYPE;
+
+import java.util.Arrays;
+
+import seedu.contax.logic.commands.FindAppointmentCommand;
+import seedu.contax.logic.parser.exceptions.ParseException;
+import seedu.contax.model.appointment.ClientNameContainsKeywordsPredicate;
+import seedu.contax.model.appointment.NameContainsKeywordsPredicate;
+
+
+
+/**
+ * Parses input arguments and creates a new FindAppointment object
+ */
+public class FindAppointmentCommandParser implements Parser<FindAppointmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindAppointment
+     * and returns a FindAppointment object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindAppointmentCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindAppointmentCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(trimmedArgs, PREFIX_SEARCH_TYPE);
+
+        String[] outputKeywords = argMultimap.getPreamble().trim().split("\\s+");
+
+        if (argMultimap.getValue(PREFIX_SEARCH_TYPE).isEmpty()) {
+            return new FindAppointmentCommand(new NameContainsKeywordsPredicate(Arrays.asList(outputKeywords)));
+        } else {
+            switch (argMultimap.getValue(PREFIX_SEARCH_TYPE).get()) {
+            case "person":
+                return new FindAppointmentCommand(new
```

Comment:
seems like you're using a command param to find the appointment by person, but i can't get it to work cos i'm not sure if i have the right command. pls update the findappointment command message with the appropriate params and syntax

Reply:
nevermind zong han told me the syntax and i tested it does work